pkg/runtime/runtime.go

```diff
@@ -1075,12 +1075,24 @@ func (a *DaprRuntime) initNameResolution() error {
 
 func (a *DaprRuntime) publishMessageHTTP(msg *pubsub.NewMessage) error {
 	var cloudEvent map[string]interface{}
+
 	err := a.json.Unmarshal(msg.Data, &cloudEvent)
 	if err != nil {
 		log.Debug(errors.Errorf("failed to deserialize cloudevent: %s", err))
 		return err
 	}
 
+	contentType := cloudEvent[pubsub.DataContentTypeField].(string)
+	if !contenttype.IsStringContentType(contentType) && !contenttype.IsJSONContentType(contentType) {
```

Comment:
The check for what is a binary here should be changed. I suggest that we add a check in components-contrib for `application/octet-stream` in a method called `contenttype.IsBinaryType(contentType)`